js/index.js

```diff
@@ -52,22 +52,57 @@ closePopupBtn.addEventListener('click', () => {
   // body.style.overflowY = 'visible';
 });
 
-const email = document.getElementById('email');
 const form = document.querySelector('.contact-form');
+const formName = document.getElementById('name');
+const formEmail = document.getElementById('email');
+const formMessage = document.getElementById('message');
 const errorMessage = document.getElementById('error-message');
 
 form.addEventListener('submit', (e) => {
   const messages = [];
   const isUpperCase = (string) => /[A-Z]/.test(string);
 
-  if (isUpperCase(email.value) === true) {
+  if (isUpperCase(formEmail.value) === true) {
     messages.push('Email must be in lowercase!');
     errorMessage.innerText = messages.join(', ');
     e.preventDefault();
   }
-  if (email.value === '' || email.value === null) {
+  if (formEmail.value === '' || formEmail.value === null) {
     messages.push('Email field can not be empty!');
     errorMessage.innerText = messages.join(', ');
     e.preventDefault();
   }
 });
+
+// Preserve data
+const formData = {
+  name: '',
+  email: '',
+  text: '',
+};
+
+function setData() {
+  const formEntry = JSON.parse(localStorage.getItem('formData'));
+  formData.name = formEntry.value;
+  formData.email = formEntry.value;
+  formData.message = formEntry.value;
+}
+
+function getData() {
+  formData.name = formName.value;
+  formData.email = formEmail.value;
+  formData.message = formMessage.value;
+  const toString = JSON.stringify(formData);
+  localStorage.setItem('formData', toString);
+  setData();
+}
+
+if (!localStorage.getItem('formData')) {
+  getData();
+} else {
+  setData();
+}
```

Comment:
Please add `onchange` attribute for your HTML form input fields that way you can save every change that the user inputs and when the page reloads it'll get those values from `local storage` and file them out. Here when I input values but before submitting when I refresh the page it's not getting those values kindly fix that.

https://user-images.githubusercontent.com/34578631/174427283-db323815-59bc-4936-a01c-71f5b41acd91.mov


Also you can check this if statement either on `window.onload = () => {...}` or on `document.addEventListener('DOMContentLoaded', () => {...})` to get the values.

Reply:
> Please add `onchange` attribute for your HTML form input fields that way you can save every change that the user inputs and when the page reloads it'll get those values from `local storage` and file them out. Here when I input values but before submitting when I refresh the page it's not getting those values kindly fix that.
> 
>  Screen.Recording.2022-06-18.at.10.07.10.AM.mov 
> 
> Also you can check this if statement either on `window.onload = () => {...}` or on `document.addEventListener('DOMContentLoaded', () => {...})` to get the values.

@Mre55 , been trying to implement. but still not working   well for me. Can you kindly help me out here. Left with this task so i finish my week.

Reply:
@Fourteen98, sorry for the late response have you manage to fix it or should I help?

Reply:
> @Fourteen98, sorry for the late response have you manage to fix it or should I help?

No, Still having some issues. @Mre55

Reply:
I was Delayed on  this pair programming task by my coding partner untill she was removed. Working solo is hard @Mre55

Reply:
Let me what you have until now.

Reply:
@Fourteen98, can we join a call?

Reply:
@Mre55  sure

Reply:
> @Mre55 sure

will be glad

Reply:
https://microverse.zoom.us/j/93685431683?pwd=dG8zRmZ2MklCSGZYeUZJdktVaGZUUT09
@Mre55

Reply:
@Mre55 ,i think it is working now, can you kindly confirm that.

Reply:
@Fourteen98, yes it's working fine you can submit now.

Reply:
W

> @Fourteen98, yes it's working fine you can submit now.

@Mre55  thanks so much,

Reply:
@Fourteen98, you're welcome.

Reply:
@Mre55 , i was turned down. says  the message has the same problem.

Reply:
@Mre55 i have reached out on slack

Reply:
@Fourteen98, I have replayed on slack